lib/reversetunnel/track/tracker.go

```diff
@@ -153,76 +151,74 @@ func (t *Tracker) StopAll() {
 func (t *Tracker) tick() {
 	t.mu.Lock()
 	defer t.mu.Unlock()
+	if t.sets == nil {
+		return
+	}
+
 	cutoff := time.Now().Add(-1 * t.ProxyExpiry)
-	for addr, set := range t.sets {
-		if set.expire(cutoff) > 0 {
-			count := len(set.proxies)
-			if count < 1 {
-				count = 1
-			}
-			t.wp.Set(addr, uint64(count))
+	if t.sets.expire(cutoff) > 0 {
+		count := len(t.sets.proxies)
+		if count < 1 {
```

Comment:
My educated guess is that setting the workpool target to anything `< 1` would cause it to close the underlying workgroup and thus require a new one to be created in the future when the target becomes `>= 1`. By instead setting the workpool target to `1` it resets the workgroup without deleting it.


@fspmarshall it looks like it has been like this since you added the `tracker` can you please correct me if I am wrong here?

Reply:
The count is the number of proxies we expect to discover, based on the heartbeats we've seen.  We don't start off knowing about any proxies since it is the first proxy that we connect to that tells us about its peers.  Therefore if we don't know about any proxies yet, we just try to find at least one, and then wait for it to tell us the real number.